swarm/src/lib.rs

```diff
@@ -324,13 +323,11 @@ where
     /// List of protocols that the behaviour says it supports.
     supported_protocols: SmallVec<[Vec<u8>; 16]>,
 
+    confirmed_external_addr: HashSet<Multiaddr>,
```

Comment:
Ah, bummer, we only need this for `Swarm::external_addresses`, right? I was hoping that there was no more need for the `Swarm` to keep this state. That said, it is not complex, thus not a big deal.

Reply:
Yes. At the moment, our tests rely on it to connect two swarms together. We could move away from it by introducing a dedicated behaviour that we use in every test but I figured that is too much churn for now.